packages/destination-actions/src/destinations/stackadapt-audiences/__tests__/onDelete.test.ts

```diff
@@ -0,0 +1,259 @@
+import Definition from '../index'
+import { SegmentEvent } from '@segment/actions-core/*'
+
+const testDestination = createTestIntegration(Definition)
+const mockGqlKey = 'test-graphql-key'
+
+const gqlHostUrl = 'https://api.stackadapt.com'
+const gqlPath = '/graphql'
+const mockUserId = 'user-id'
+const mockAdvertiserId = '23'
+const mockMappings = { advertiser_id: mockAdvertiserId }
+
+const deleteEventPayload: Partial<SegmentEvent> = {
+  userId: mockUserId,
+  type: 'identify',
+  context: {
+    personas: {
+      computation_class: 'audience',
+      computation_key: 'first_time_buyer',
+      computation_id: 'aud_123'
+    }
+  }
+}
+
+describe('onDelete action', () => {
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  it('should delete a profile successfully', async () => {
+    let deleteRequestBody
+
+    nock(gqlHostUrl)
+      .post(gqlPath, (body) => {
+        return body
+      })
+      .reply(200, {
+        data: {
+          tokenInfo: {
+            scopesByAdvertiser: {
+              nodes: [
+                {
+                  advertiser: {
+                    id: mockAdvertiserId
+                  }
+                }
+              ]
+            }
+          }
+        }
+      })
+
+    nock(gqlHostUrl)
+      .post(gqlPath, (body) => {
+        deleteRequestBody = body
+        return body
+      })
+      .reply(200, {
+        data: {
+          deleteProfilesWithExternalIds: {
+            userErrors: []
+          }
+        }
+      })
+
+    const event = createTestEvent({
+      userId: mockUserId,
+      type: 'identify',
+      context: {
+        personas: {
+          computation_class: 'audience',
+          computation_key: 'first_time_buyer',
+          computation_id: 'aud_123'
+        }
+      }
+    })
+
+    const responses = await testDestination.testAction('onDelete', {
+      event,
+      useDefaultMappings: true,
+      mapping: { userId: mockUserId },
+      settings: { apiKey: mockGqlKey }
+    })
+
+    // Assert that two responses were received, one for the token query and one for the profile deletion
+    expect(responses.length).toBe(2)
+
+    // Ensure the second request body (profile deletion) matches the expected mutation
+    expect(deleteRequestBody).toMatchInlineSnapshot(`
+      Object {
+        "query": "mutation {
+          deleteProfilesWithExternalIds(
+            externalIds: [\\"${mockUserId}\\"],
+            advertiserIDs: [\\"${mockAdvertiserId}\\"],
+            externalProvider: \\"segmentio\\"
+          ) {
+            userErrors {
+              message
+              path
+            }
+          }
+        }",
+      }
+    `)
+  })
+
+  it('should throw error if no advertiser ID is found', async () => {
+    nock(gqlHostUrl)
+      .post(gqlPath)
+      .reply(200, {
+        data: {
+          tokenInfo: {
+            scopesByAdvertiser: {
+              nodes: []
+            }
+          }
+        }
```

Comment:
I noticed that this shows up multiple times in the test. You can refactor it into a function and pass in the `nodes` part as input to reduce duplication

Reply:
@illumin04 you might have missed this. Also applies to `deleteProfilesWithExternalIds ` query in test